pkg/api/vlabs/validate.go

```diff
@@ -401,6 +415,18 @@ func (a *Properties) validateAgentPoolProfiles() error {
 			if a.AgentPoolProfiles[i].AvailabilityProfile != a.AgentPoolProfiles[0].AvailabilityProfile {
 				return errors.New("mixed mode availability profiles are not allowed. Please set either VirtualMachineScaleSets or AvailabilitySet in availabilityProfile for all agent pools")
 			}
+
+			if a.AgentPoolProfiles[i].AvailabilityProfile == AvailabilitySet {
+				if a.AgentPoolProfiles[i].AvailabilityZones != nil && len(a.AgentPoolProfiles[i].AvailabilityZones) > 0 {
+					return errors.New("Availability Zones are not supported with an AvailabilitySet. Please either remove availabilityProfile or set availabilityProfile to VirtualMachineScaleSets")
+				}
+			}
+
+			if a.AgentPoolProfiles[i].AvailabilityZones != nil && len(a.AgentPoolProfiles[i].AvailabilityZones) > 0 {
+				if a.AgentPoolProfiles[i].Count < len(a.AgentPoolProfiles[i].AvailabilityZones)*2 {
+					return errors.New("The node count and the number of availability zones provided can result in zone imbalance. To achieve zone balance, each zone should have at least 2 nodes or more. ")
+				}
+			}
```

Comment:
can we add a validation for singlePlacementGroups for VMSS only?

Reply:
done